language-plutus-core/src/Language/PlutusCore/Rename/Internal.hs

```diff
@@ -146,11 +146,8 @@ renameTypeM (TyLam ann name kind ty)    =
     withFreshenedName name $ \nameFr -> TyLam ann nameFr kind <$> renameTypeM ty
 renameTypeM (TyForall ann name kind ty) =
     withFreshenedName name $ \nameFr -> TyForall ann nameFr kind <$> renameTypeM ty
-renameTypeM (TyIFix ann pat arg)        = TyIFix ann <$> renameTypeM pat <*> renameTypeM arg
-renameTypeM (TyApp ann fun arg)         = TyApp ann <$> renameTypeM fun <*> renameTypeM arg
-renameTypeM (TyFun ann dom cod)         = TyFun ann <$> renameTypeM dom <*> renameTypeM cod
-renameTypeM (TyVar ann name)            = TyVar ann <$> renameNameM name
-renameTypeM ty@TyBuiltin{}              = pure ty
+renameTypeM x                           =
+    (traverseOf typeSubtypes renameTypeM >=> traverseOf typeTyVars renameNameM) x
```

Comment:
Same considerations as with `renameTermM`. If we can have a speedup by not turning several straightforward lines of code into one line, why not do that? Also, I think with sufficient cleverness we could handle binders generally as well, but I'd rather avoid such cleverness. Also, if we add a binder to the language (we talked about existential quantification previously), we won't have a warning here and the binder won't be handled correctly.

Reply:
I guess it's a matter of taste what you think is nicer. I think it's nice to say "in the other cases: just look at the subtypes and type vars, whatever they are", but I certainly take your point about the pattern matching.